.github/workflows/openssl3.yml

```diff
@@ -0,0 +1,91 @@
+name: openssl3
+
+on:
+  push:
+    branches: [ 'main' ]
+    paths: ['.github/workflows/openssl3.yml', 'openssl3/**']
+  pull_request:
+    branches: [ 'main' ]
+    paths: ['.github/workflows/openssl3.yml', 'openssl3/**']
+  workflow_call:
+    inputs:
+      build_main:
+        description: "Build using liboqs and oqsprovider main branches"
+        required: false
+        default: false
+        type: boolean
+      release_tag:
+        description: "Which docker tag to push to"
+        required: false
+        type: string
+  workflow_dispatch:
+    inputs:
+      build_main:
+        description: "Build using liboqs and oqsprovider main branches"
+        required: false
+        default: false
+        type: boolean
+      release_tag:
+        description: "Which docker tag to push to"
+        required: false
+        type: string
+
+env:
+  build-args: |
+    LIBOQS_TAG=main
+    OQSPROVIDER_TAG=main
+  push: ${{ github.repository == 'open-quantum-safe/oqs-demos' && github.ref == 'refs/heads/main' && github.event_name != 'pull_request' }}
```

Comment:
As mentioned in my previous comment, this is the line that determines when the workflow should push a release.

Currently this is when the workflow is run on the main branch of the original (non-fork) repo and is not a PR.

@baentsch I was wondering if this should instead be handled as a input instead. Meaning the workflow would only push when run manually with a new `push_release` input set to true. If we still wanted to have it push automatically in some situations we could add other automated triggers for those situations

Reply:
Sounds like a sensible approach to me.

Reply:
Which approach are you calling "sensible"? The one I've already coded (# 1) or my suggestion in the comment (# 2)

1. run on the main branch of the original (non-fork) repo and is not a PR
2. only push when run manually with a new push_release input set to true